core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -121,35 +121,45 @@ public FlowIdentifier removeInstances(Stack stack, Collection<String> instanceId
             throw new BadRequestException("Downscale request cannot process a 'null' instanceIds collection");
         }
 
-        Map<String, Set<Long>> instanceIdsByHostgroupMap = new HashMap<>();
-        for (String instanceId : instanceIds) {
-            // TODO CB-14929: CB-15154 An invalid instanceId should not necessarily fail a downscale-stop operation?
-            InstanceMetaData metaData = updateNodeCountValidator.validateInstanceForDownscale(instanceId, stack);
-            instanceIdsByHostgroupMap.computeIfAbsent(metaData.getInstanceGroupName(), s -> new LinkedHashSet<>()).add(metaData.getPrivateId());
-        }
-        updateNodeCountValidator.validateServiceRoles(stack, instanceIdsByHostgroupMap.entrySet()
-                .stream()
-                .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().size() * -1)));
-        LOGGER.info("ZZZ: removeInstances: instanceIds/hostnames received: {}", instanceIds);
-        LOGGER.info("ZZZ: removeInstances: computedInstanceIds: {}", instanceIdsByHostgroupMap);
-        // TODO CB-14929: Validations for start-stop
-        if (!forced) {
-            for (Map.Entry<String, Set<Long>> entry : instanceIdsByHostgroupMap.entrySet()) {
-                String instanceGroupName = entry.getKey();
-                int scalingAdjustment = entry.getValue().size() * -1;
-                updateNodeCountValidator.validateScalabilityOfInstanceGroup(stack, instanceGroupName, scalingAdjustment);
-                updateNodeCountValidator.validateScalingAdjustment(instanceGroupName, scalingAdjustment, stack);
-            }
-        }
-        if (scalingStrategy != null) {
-            // TODO: CB-14929: CB-15154 stopstart can only support a single hostGroup being specified.
-            // TODO: CB-14929: Should this not be putting the stack into a UpdateInProgress state?, and maybe run inside a transaction? (similar to the upscale)
-            if (instanceIdsByHostgroupMap.size() > 1) {
-                throw new BadRequestException("Downscale via Instance Stop cannot process more than one host group");
+        try {
```

Comment:
Was not sure how this change should be handled. Looking at the rest of the code ... very few snippets are inside a transaction, and update the state in the DB within StackOperationsService.java ... my guess is this state gets updated further down when the operation actually starts.
The question remains if this is the right place to make this change.

Reply:
Even I am not sure if this is how it is supposed to be. I have just replicated what is done for a normal scale down.

Reply:
@sodre90, @lnardai - do you have an opinion on this.
Most of StackOperationsService does not use a transaction (transactionService.required()) to wrap the validations, potentially state changes, etc that they perform. updateNodeCount is in fact the only one which uses this.
Question1: Does any of this need to be in a transaction, and why - if it does? 
Question2: Some operations in the same class update the stack/cluster-state inline. Others do not (e.g. updateNodeCount sets the state to UPSCALE_REQUESTED. However removeInstance / removeInstances do not perform any kind of status update here (Guessing the state eventually updates once the state machine starts executing). Any insights on how this should be handled?